daemon/api_debug.go

```diff
@@ -345,16 +345,29 @@ func createRecovery(st *state.State, label string) Response {
 }
 
 type featureResponse struct {
-	Tasks      []string              `json:"tasks"`
+	Tasks      []taskResponse        `json:"tasks"`
 	Interfaces []string              `json:"interfaces"`
 	Endpoints  []featureEndpoint     `json:"endpoints"`
 	Changes    []string              `json:"changes"`
 	Ensures    []swfeats.EnsureEntry `json:"ensures"`
 }
 
+type taskResponse struct {
+	Kind    string `json:"kind"`
+	HasUndo bool   `json:"has-undo,omitempty"`
+}
+
 func getFeatures(c *Command) Response {
 	runner := c.d.overlord.TaskRunner()
-	tasks := runner.KnownTaskKinds()
+	taskKinds := runner.KnownTaskKinds()
+	taskResponses := make([]taskResponse, 0, len(taskKinds))
+	for _, taskKind := range taskKinds {
+		t := taskResponse{
+			Kind:    taskKind,
+			HasUndo: c.d.overlord.TaskRunner().TaskKindHasUndo(taskKind),
```

Comment:
Is there a reason not to use the existing `runner` variable here, as is done elsewhere?

```suggestion
			HasUndo: runner.TaskKindHasUndo(taskKind),
```